pkg/machine/wsl/machine.go

```diff
@@ -0,0 +1,1119 @@
+
+	backoff := 500 * time.Millisecond
+	for i := 0; i < 5; i++ {
+		err = runCmdPassThroughTee(log, "wsl", "--update")
+		if err == nil {
+			break
+		}
+		// In case of unusual circumstances (e.g. race with installer actions)
+		// retry a few times
+		message = "An error occured attempting the WSL Kernel update, retrying..."
+		fmt.Println(message)
+		fmt.Fprintln(log, message)
+		time.Sleep(backoff)
+		backoff *= 2
+	}
+
+	if err != nil {
+		return errors.Wrap(err, "could not install WSL Kernel")
+	}
+
+	return nil
+}
+
+func getElevatedOutputFileName() (string, error) {
+	dir, err := homedir.GetDataHome()
+	if err != nil {
+		return "", err
+	}
+	return filepath.Join(dir, "podman-elevated-output.log"), nil
+}
+
+func dumpOutputFile() {
+	file, err := getElevatedOutputFileRead()
+	if err != nil {
+		logrus.Debug("could not find elevated child output file")
+		return
+	}
+	defer file.Close()
+	_, _ = io.Copy(os.Stdout, file)
+}
+
+func getElevatedOutputFileRead() (*os.File, error) {
+	return getElevatedOutputFile(os.O_RDONLY)
+}
+
+func getElevatedOutputFileWrite() (*os.File, error) {
+	return getElevatedOutputFile(os.O_WRONLY | os.O_CREATE | os.O_APPEND)
+}
+
+func appendOutputIfError(write bool, err error) {
+	if write && err == nil {
+		return
+	}
+
+	if file, check := getElevatedOutputFileWrite(); check == nil {
+		defer file.Close()
+		fmt.Fprintf(file, "Error: %v\n", err)
+	}
+}
+
+func truncateElevatedOutputFile() error {
+	name, err := getElevatedOutputFileName()
+	if err != nil {
+		return err
+	}
+
+	return os.Truncate(name, 0)
+}
+
+func getElevatedOutputFile(mode int) (*os.File, error) {
+	name, err := getElevatedOutputFileName()
+	if err != nil {
+		return nil, err
+	}
+
+	dir, err := homedir.GetDataHome()
+	if err != nil {
+		return nil, err
+	}
+
+	if err = os.MkdirAll(dir, 0755); err != nil {
+		return nil, err
+	}
+
+	return os.OpenFile(name, mode, 0644)
+}
+
+func isMsiError(err error) bool {
+	if err == nil {
+		return false
+	}
+
+	if eerr, ok := err.(*exec.ExitError); ok {
+		switch eerr.ExitCode() {
+		case 0:
+			fallthrough
+		case ErrorSuccessRebootInitiated:
+			fallthrough
+		case ErrorSuccessRebootRequired:
+			return false
+		}
+	}
+
+	return true
+}
+func toDist(name string) string {
+	if !strings.HasPrefix(name, "podman") {
+		name = "podman-" + name
+	}
+	return name
+}
+
+func withUser(s string, user string) string {
+	return strings.ReplaceAll(s, "[USER]", user)
+}
+
+func runCmdPassThrough(name string, arg ...string) error {
+	logrus.Debugf("Running command: %s %v", name, arg)
+	cmd := exec.Command(name, arg...)
+	cmd.Stdin = os.Stdin
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	return cmd.Run()
+}
+
+func runCmdPassThroughTee(out io.Writer, name string, arg ...string) error {
+	logrus.Debugf("Running command: %s %v", name, arg)
+
+	// TODO - Perhaps improve this with a conpty pseudo console so that
+	//        dism installer text bars mirror console behavior (redraw)
+	cmd := exec.Command(name, arg...)
```

Comment:
nit: Following exec pattern is being used at multiple places could we convert it to utility function

Reply:
There are some variations in each of these that could lead to a long common multi-argument method call with multiple flags and inputs when consolidated. Not sure if that improves readability.